app/helpers/checkout_helper.rb

```diff
@@ -40,6 +40,24 @@ def checkout_adjustments_for(order, opts = {})
       )
     end
 
+    tax_rate_adjustments = adjustments.select { |a|
+      a.originator_type == "Spree::TaxRate" && a.adjustable_type == "Spree::Adjustment"
+    }
+    adjustments.reject! { |a|
+      a.originator_type == "Spree::TaxRate" && a.adjustable_type == "Spree::Adjustment"
+    }
```

Comment:
Would this be easier?
```suggestion
    tax_rate_adjustments, adjustments = adjustments.partition { |a|
      a.originator_type == "Spree::TaxRate" && a.adjustable_type == "Spree::Adjustment"
    }
```


And if we defined a new method on adjustments:

```rb
    tax_rate_adjustments, adjustments = adjustments.partition(&:adjustment_tax?)
```

Reply:
I do like that suggestion and committed it. I'm slightly wary of having a new method on adjustments since... well, I guess because I don't feel familiar with this part of the code to know for certain that the conditional we use here is what you'd want in all cases where you might ask "is this a tax adjustment". Which is sort of like saying I don't have a sense of what the right name for it would be, so for now I'm more inclined to leave it as the explicit conditional until I do :)